tools/TimeFileMaker.py

```diff
@@ -0,0 +1,181 @@
+    partial_seconds = int(100 * (seconds - full_seconds))
+    return sign + '%dm%02d.%02ds' % (minutes, full_seconds, partial_seconds)
+
+def sum_times(times, signed=False):
+    '''
+    Takes the values of an output from get_times, parses the time
+    strings, and returns their sum, in the same string format.
+    '''
+    return from_seconds(sum(map(to_seconds, times)), signed=signed)
+
+def format_percentage(num, signed=True):
+    sign = ('-' if num < 0 else '+') if signed else ''
+    num = abs(num)
+    whole_part = int(num * 100)
+    frac_part = int(100 * (num * 100 - whole_part))
+    return sign + '%d.%02d%%' % (whole_part, frac_part)
+
+def make_diff_table_string(left_times_dict, right_times_dict,
+                           descending=True,
+                           left_tag="After", tag="File Name", right_tag="Before", with_percent=True,
+                           change_tag="Change", percent_change_tag="% Change"):
+    # We first get the names of all of the compiled files: all files
+    # that were compiled either before or after.
+    all_names_dict = dict()
+    all_names_dict.update(right_times_dict)
+    all_names_dict.update(left_times_dict) # do the left (after) last, so that we give precedence to those ones
+    prediff_times = tuple((name, to_seconds(left_times_dict.get(name,'0m0.0s')), to_seconds(right_times_dict.get(name,'0m0.0s')))
+                          for name in all_names_dict.keys())
+    diff_times_dict = dict((name, from_seconds(lseconds - rseconds, signed=True))
+                           for name, lseconds, rseconds in prediff_times)
+    percent_diff_times_dict = dict((name, ((format_percentage((lseconds - rseconds) / rseconds))
+                                           if rseconds != 0 else (INFINITY if lseconds > 0 else 'N/A')))
+                                   for name, lseconds, rseconds in prediff_times)
+    # update to sort by approximate difference, first
+    get_key = make_sorting_key(all_names_dict, descending=descending)
+    all_names_dict = dict((name, (abs(int(to_seconds(diff_times_dict[name]))), get_key(name)))
+                          for name in all_names_dict.keys())
+    names = sorted(all_names_dict.keys(), key=all_names_dict.get)
+    #names = get_sorted_file_list_from_times_dict(all_names_dict, descending=descending)
+    # set the widths of each of the columns by the longest thing to go in that column
+    left_sum = sum_times(left_times_dict.values())
+    right_sum = sum_times(right_times_dict.values())
+    left_sum_float = sum(map(to_seconds, left_times_dict.values()))
+    right_sum_float = sum(map(to_seconds, right_times_dict.values()))
+    diff_sum = from_seconds(left_sum_float - right_sum_float, signed=True)
+    percent_diff_sum = (format_percentage((left_sum_float - right_sum_float) / right_sum_float)
+                        if right_sum_float > 0 else 'N/A')
+    left_width = max(max(map(len, ['N/A'] + list(left_times_dict.values()))), len(left_sum))
```

Comment:
I don't understand your comment.  The inner `max` is using the first form, `max(iterable)`, while the outer `max` is using the second form, `max(arg1, arg2)`; these two forms do not mix; there is no `max(iterable1, arg2)` form.